apps/src/aichat/types.ts

```diff
@@ -132,6 +132,8 @@ export enum Visibility {
 export interface LevelAichatSettings {
   initialCustomizations: AiCustomizations;
   visibilities: {[key in keyof AiCustomizations]: Visibility};
+  // This system prompt is hidden from students and adds additional safety features or hidden guidelines to a level.
+  levelSystemPrompt: string;
```

Comment:
So we still end up passing this to the front end, yeah (even though we're not using it there any more)? I assume we need to keep this because this type is used on the levelbuilder side?

I guess that's a bit of a downside of using the same type in the levelbuilder editing experience and what teachers/students see, but something if we can figure out down the line if we need to have more separation between levelbuilders can customize and what we end up using for teachers/students.

Reply:
Yeah, FWIW though even if we didn't add this type we'd still be passing it to the front end since we currently just send the whole "properties" object via ajax call. Though in the future to your point we could add some more abstraction and send a different slice of the properties object to lab2 labs, and keep certain fields levelbuilder-editable only.

Reply:
Interesting points - thanks! I wonder if`levelSystemPrompt` a unique case or if there are other fields that are editable by levelbuilders but not necessarily used on the frontend.